weave

```diff
@@ -700,17 +711,19 @@ wait_for_status() {
     done
 }
 
-# Wait until container $1 outputs a line containing $2 in its log
-wait_for_log() {
-    fifo=/tmp/tmpfifo.$$
-    mkfifo $fifo || exit 1
-    docker logs -f $1 >$fifo 2>&1 &
-    res=0
-    grep -q "$2" $fifo || res=1
-    kill $! 2>/dev/null || true
-    rm $fifo
-    [ $res -eq 1 ] && echo "$(death_msg $1)" >&2
-    return $res
+# Wait until container $1 is alive enough to respond to "GET /status"
+# http request is to unix socket "status.sock", we assume the container
+# has curl >= 7.40
+wait_for_status_unix() {
+    err_msg=$(death_msg $1)
+    while true ; do
+        http_call_unix $1 status.sock GET /status >/dev/null 2>&1 && return 0
+        if ! container_id $1 >/dev/null 2>&1 ; then
+            echo "$err_msg" >&2
+            return 1
+        fi
+        fractional_sleep 3.1
+    done
```

Comment:
Surely there must be a way to refactor `wait_for_status_unix` and `http_call_unix` to avoid all the duplication with `wait_for_status` and `http_call`.